solr/core/src/java/org/apache/solr/cloud/ElectionContext.java

```diff
@@ -63,7 +64,7 @@
   final ZkNodeProps leaderProps;
   final String id;
   final String leaderPath;
-  volatile String leaderSeqPath;
+  final AtomicReference<String> leaderSeqPath = new AtomicReference<>();
```

Comment:
Good catch! Fixed. Turns out that the new test had a bad assertion thanks to the same problem -- looking at the `AtomicReference` instead of its string value. Doh! :(